### PR TITLE
Expose missing Node API

### PR DESCRIPTION
**Pull request by dkozykowski**
### Details:
 - Add bindings for following methods:
 
- [x] ```size_t get_input_size() const```
- [x] ```std::vector<Output<Node>> input_values() const```
- [x] ```Output<Node> input_value(size_t input_index) const```
- [x] ```descriptor::Tensor& get_input_tensor(size_t i) const```
- [x] ```virtual bool evaluate(const ov::runtime::TensorVector& output_values,
 const ov::runtime::TensorVector& input_values,
 const EvaluationContext& evaluationContext) const```
- [x] ```virtual bool evaluate(const ov::runtime::TensorVector& output_values,
 const ov::runtime::TensorVector& input_values) const```

### Tickets:
 - 67431

### CODE REVIEW

File: runtime/bindings/python/src/pyopenvino/graph/node.cpp [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #include <pybind11/stl_bind.h>
 
 #include "dict_attribute_visitor.hpp"
+#include "openvino/core/node.hpp"
```
Comment (jiwaszki) [inline runtime/bindings/python/src/pyopenvino/graph/node.cpp:12]: Please remove it from line 5 or keep it there and remove this one. Isn't that the same include?

File: runtime/bindings/python/src/pyopenvino/graph/node.cpp [hunk 2]
```diff
@@ -91,14 +155,51 @@ void regclass_graph_Node(py::module m) {
...
+    node.def("input_values",
+             &ov::Node::input_values,
+             R"(
+                 Returns list of inputs to the node, in order
+
+                 Returns
+                 ----------
+                 inputs : List[Input]
+                    List of node's inputs
+
```
Comment (jiwaszki) [inline runtime/bindings/python/src/pyopenvino/graph/node.cpp:167]: Remove empty line.

File: runtime/bindings/python/src/pyopenvino/graph/node.cpp [hunk 3]
```diff
@@ -91,14 +155,51 @@ void regclass_graph_Node(py::module m) {
                 get_element_type : Type
                     Type of the output.
              )");
+    node.def("input_values",
+             &ov::Node::input_values,
+             R"(
+                 Returns list of inputs to the node, in order
```
Comment (jiwaszki) [inline runtime/bindings/python/src/pyopenvino/graph/node.cpp:161]:
```suggestion
                 Returns list of node's inputs, in order.
```

File: runtime/bindings/python/src/pyopenvino/graph/node.cpp [hunk 4]
```diff
@@ -80,7 +81,70 @@ void regclass_graph_Node(py::module m) {
...
+                Returns
+                ----------
+                evaluate : bool
+             )");
+    node.def("get_input_tensor",
+             &ov::Node::get_input_tensor,
+             py::arg("i"),
+             py::return_value_policy::reference_internal,
+             R"(
+                Returns the tensor for the node input with index i
```
Comment (jiwaszki) [inline runtime/bindings/python/src/pyopenvino/graph/node.cpp:136]:
```suggestion
                Returns the tensor for the node's input with index i
```

File: runtime/bindings/python/src/pyopenvino/graph/node.cpp [hunk 5]
```diff
@@ -80,7 +81,70 @@ void regclass_graph_Node(py::module m) {
...
+            return self.evaluate(output_values, input_values);
+        },
+        py::arg("output_values"),
+        py::arg("input_values"),
+        R"(
+                Evaluate the function on inputs, putting results in outputs
+                Parameters
+                ----------
+                output_tensors : List[op.Tensor]
+                    Tensors for the outputs to compute. One for each result
```
Comment (jiwaszki) [inline runtime/bindings/python/src/pyopenvino/graph/node.cpp:124]:
```suggestion
                    Tensors for the outputs to compute. One for each result.
```

File: runtime/bindings/python/tests/test_ngraph/test_basic.py [hunk 6]
```diff
@@ -361,6 +363,67 @@ def test_node_output():
...
+        [data1, data2]
+    )
+
+def test_node_input_tensor():
+    data1 = np.array([1, 2, 3])
+    data2 = np.array([3, 2, 1])
+
+    node = ops.add(data1, data2)
+
+    assert(isinstance(node.get_input_tensor(0), descriptor.Tensor))
```
Comment (jiwaszki) [inline runtime/bindings/python/tests/test_ngraph/test_basic.py:405]: Can you first assign it somewhere and then assert? Just curious about behavior.

**Review by jiwaszki — CHANGES_REQUESTED** [review]

File: runtime/bindings/python/src/pyopenvino/graph/node.cpp (hunk 1 above)
Comment (dkozykowski) [reply runtime/bindings/python/src/pyopenvino/graph/node.cpp]: It is, however it was not my change. Removed redundant include.

File: runtime/bindings/python/tests/test_ngraph/test_basic.py (hunk 6 above)
Comment (dkozykowski) [reply runtime/bindings/python/tests/test_ngraph/test_basic.py]: Sure, done.

File: src/bindings/python/tests/test_ngraph/test_basic.py [hunk 7]
```diff
@@ -4,6 +4,8 @@
 import json
 
 import numpy as np
+from numpy.core.numeric import _ones_like_dispatcher
+from numpy.lib.arraysetops import isin
```
Comment (jiwaszki) [inline src/bindings/python/tests/test_ngraph/test_basic.py:8]: Are these two required? It seems like unused imports.

**Review by jiwaszki — COMMENTED** [review]
Fix flake issues and it is ready to merge.

File: src/bindings/python/tests/test_ngraph/test_basic.py (hunk 7 above)
Comment (dkozykowski) [reply src/bindings/python/tests/test_ngraph/test_basic.py]: Good catch, removed redundant imports.

**Review by sdurawa — APPROVED** [review]

**Review by tomdol — APPROVED** [review]

**Pull request merged**